lingua_franca/lang/parse_en.py

```diff
@@ -587,18 +587,43 @@ def extract_duration_en(text):
         'days': 0,
         'weeks': 0
     }
+    units = ['months', 'years', 'decades', 'centurys', 'millenniums'] + \
+            list(time_units.keys())
 
     pattern = r"(?P<value>\d+(?:\.?\d+)?)(?:\s+|\-){unit}s?"
     text = _convert_words_to_numbers_en(text)
+    text = text.replace("centuries", "century").replace("millenia",
+                                                        "millennium")
+    text = text.replace("a day", "1 day").replace("a year", "1 year") \
```

Comment:
this might be more readable as:

```python
singular_durations = ('day', 'month', 'year', 'decade', 'century', 'millenium')
for word in singular_durations:
    text = text.replace(f'a {word}', '1 word')
```